serverlessworkflow/modules/ROOT/pages/eventing/consume-producing-events-with-kafka.adoc

```diff
@@ -1 +1,89 @@
+
+== Adding Kafka dependency
+
+This section explains how to instruct Kogito Serverless Workflow to use the Kafka Smallrye connector. This info is extracted from link:{kogito_kafka_integration_url}[Kogito documentation]
+
+Although Kogito Messaging capabilities are optional, they are already included the Quarkus Serverless Workflow Extension. Thats why you do not need to explicitly include the messaging add-on dependency when using Serverless Workflow. However, you need to indicate that you are going to use Kafka by adding the Kafka Quarkus Smallrye connector dependency. 
+
+If using Maven, this means including following snippet in pom.xml:
+
+[source,json]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-smallrye-reactive-messaging-kafka</artifactId>
+</dependency>
+----
+
+== Configuring Smallrye channels
+
+This section explains how to configure Smallrye channels for a Kogito Serverless Workflow using event definitions. This info is extracted from link:{kogito_messaging_url}[Kogito documentation]
+
+Channels are defined by using link:{quarkus_config_url}[Quarkus configuration]. The pattern for channel properties is `mp.messaging.[incoming|outgoing].<channel name>.<property_name>`.
+
+Kogito serverless workflow allows different channel mapping strategies:
+
+* Define one default incoming channel to receive all the incoming messages and one default outgoing channel to store all the published messages.
+
+* Define a channel for each link:{cloud_events_url}[Cloud Event] Type so that every message type has a dedicated channel.
+
+* Define a channel for certain Cloud Event Types and let the not mapped Cloud Event types to use the default incoming or outgoing channel.
+
+Kogito first searches for channel name equals to Cloud Event type in the properties. If found, it uses that channel for that Cloud Event type. If not, it searches for default channel definition. If also not existing, then an error will be reported.
+
+The name for the default incoming channel is `kogito_incoming_stream` and for the default outgoing channel is `kogito_outgoing_stream`.
+
+You can change the default incoming and outgoing topic name as follows:
+
+[source,properties]
+----
+kogito.addon.messaging.incoming.defaultName=<default channel name>
+kogito.addon.messaging.outgoing.defaultName=<default channel name>
+----
+
+This is the full list of link:{kafka_connector_properties_url}[properties] you can use for a channel when using Kafka connector.
+
+There are two mandatory properties that you need to setup for every channel defined:
+
+* `connector`, which has to be set to `smallrye-kafka`
+* Depending on the channel being input or output: 
+** if incoming, `value.deserializer`. Unless you have specific marshaling needs, will be set to `org.apache.kafka.common.serialization.ByteArrayDeserializer` or `org.apache.kafka.common.serialization.StringDeserializer`
```

Comment:
This implies that it will be set automatically, at least to me. AFAIK a user still needs to provide it, right?

Reply:
yes, you are right, is confusing, changing